tests/projects/projects_namespace_test.py

```diff
@@ -1,11 +1,17 @@
 from flask import json
+from flask.testing import FlaskClient
+from pytest_mock import MockFixture
 
 
-def test_list_should_return_nothing(client):
-    """Should return an empty array"""
+def test_list_all_elements(client: FlaskClient, mocker: MockFixture):
```

Comment:
No need to have comments here, you can change the method name as follows:

```
def test_endpoint_returns_all_elements_found_by_dao(client: FlaskClient, mocker: MockFixture):
    elements_found_by_dao = []
    from time_tracker_api.projects.projects_namespace import project_dao
    repository_find_all_mock = mocker.patch.object(project_dao.repository, 'find_all', return_value=elements_found_by_dao)

    response = client.get("/projects", follow_redirects=True)

    assert 200 == response.status_code
    
    json_data = json.loads(response.data)
    assert elements_found_by_dao == json_data
    repository_find_all_mock.assert_called_once()
```

Reply:
That is basically the JUni4 of writing code, which is pretty much discouraged. The JUnit 5 way of describing test functions with annotations is a good prove of it. The function name is just a unique container, but not a good way of providing a verbose description of the test.